api/src/auth/api_key_auth.py

```diff
@@ -58,11 +45,46 @@ def verify_token(token: str) -> dict:
     # in your API, you don't need to check each
     # one in order to figure out which was actually used
     flask.g.current_user = user
-    flask.g.current_user_log_attributes = user.get_user_log_attributes()
+
+    # Add the "username" to all logs for the rest of the request lifecycle
+    add_extra_data_to_current_request_logs({"auth.username": user.username})
 
     logger.info("Authentication successful")
 
-    return user.as_dict()
+    return user
+
+
+def _get_auth_users() -> dict[str, User] | None:
```

Comment:
How about making this into an initializer function that's called when the server starts. Then it could raise an error at startup if no list of users is configured, instead of the error happening later at request time.

Rough idea:

```python
    def create_app() -> APIFlask:
        ...
        api_key_auth.initialize(os.getenv("API_AUTH_TOKEN", None))
```

```python
    def initialize(api_auth_token: str):
        if not api_auth_token:
            raise ValueError("Missing or empty api auth token")
        ...
```

Reply:
We could do that, it would mean that the app won't startup in a new environment if we haven't setup something in parameter store, which may be desired? Do we have any docs for setting up a new environment and the variables that are required to make it work

Reply:
I don't think we have that documented yet. I like the approach a "fast fail" of not starting when any setting is missing, but I could see sometimes it's better to have a partially working system.

Reply:
Tinkering with this a bit, I do agree that doing the initialize approach is better, but unfortunately it requires a bit of restructuring of the test setup to make work. As this isn't a long-term solution and we'll scrap this for an entirely different auth approach  in the next few months, any objection with leaving it like this for now?

Reply:
No objection from me, let's merge and come back to this another time.